numpy/core/tests/test_multiarray.py

```diff
@@ -1697,60 +1697,60 @@ def test_sort(self):
-        for dtype in [np.int8, np.int16, np.int32, np.int64, np.float16, np.float32, np.float64, np.longdouble]:
-            a = np.arange(-50, 51, dtype=dtype)
-            b = a[::-1].copy()
-            for kind in self.sort_kinds:
-                msg = "scalar sort, kind=%s, dtype=%s" % (kind, dtype)
-                c = a.copy()
-                c.sort(kind=kind)
-                assert_equal(c, a, msg)
-                c = b.copy()
-                c.sort(kind=kind)
-                assert_equal(c, a, msg)
+    # all c scalar sorts use the same code with different types
+    # so it suffices to run a quick check with one type. The number
+    # of sorted items must be greater than ~50 to check the actual
+    # algorithm because quick and merge sort fall over to insertion
+    # sort for small arrays.
+
+    @pytest.mark.parametrize('dtype', [np.uint8, np.uint16, np.uint32, np.uint64, np.float16, np.float32, np.float64,
+                                       np.longdouble])
+    def test_sort_unsigned(self, dtype):
+        a = np.arange(101, dtype=dtype)
+        b = a[::-1].copy()
+        for kind in self.sort_kinds:
+            msg = "scalar sort, kind=%s, dtype=%s" % (kind, dtype)
+            c = a.copy()
+            c.sort(kind=kind)
+            assert_equal(c, a, msg)
+            c = b.copy()
+            c.sort(kind=kind)
+            assert_equal(c, a, msg)
 
-        # test complex sorts. These use the same code as the scalars
-        # but the compare function differs.
-        ai = a*1j + 1
-        bi = b*1j + 1
+    @pytest.mark.parametrize('dtype',
+                             [np.int8, np.int16, np.int32, np.int64, np.float16, np.float32, np.float64, np.longdouble])
+    def test_sort_signed(self, dtype):
+        a = np.arange(-50, 51, dtype=dtype)
+        b = a[::-1].copy()
         for kind in self.sort_kinds:
-            msg = "complex sort, real part == 1, kind=%s" % kind
-            c = ai.copy()
+            msg = "scalar sort, kind=%s, dtype=%s" % (kind, dtype)
+            c = a.copy()
             c.sort(kind=kind)
-            assert_equal(c, ai, msg)
-            c = bi.copy()
+            assert_equal(c, a, msg)
+            c = b.copy()
             c.sort(kind=kind)
-            assert_equal(c, ai, msg)
-        ai = a + 1j
-        bi = b + 1j
+            assert_equal(c, a, msg)
+
+    @pytest.mark.parametrize('dtype',
+                             [np.int8, np.int16, np.int32, np.int64, np.float16, np.float32, np.float64, np.longdouble])
+    @pytest.mark.parametrize('message, factor, number', [('real', 1j, 1), ('imag', 1, 1j)])
+    def test_sort_complex(self, message, factor, number, dtype):
+        # test complex sorts. These use the same code as the scalars
+        # but the compare function differs.
+        a = np.arange(-50, 51, dtype=dtype)
+        b = a[::-1].copy()
+        ai = a * factor + number
+        bi = b * factor + number
```

Comment:
```suggestion
        ai = (a * (1+1j)).astype(dtype)
        bi = (b * (1+1j)).astype(dtype)
        setattr(ai, part, 1)
        setattr(bi, part, 1)
```

Reply:
are you sure about this change? it results in  numpy.ComplexWarning: Casting complex values to real discards the imaginary part

Reply:
oh whoops. That needs to be `.astype(the_complex_version_of_dtype)`. Maybe just add the complex type to the `pytest.mark.parametrize`?